server/jetstream_api.go

```diff
@@ -1173,11 +1173,14 @@ func (s *Server) jsStreamCreateRequest(sub *subscription, c *client, subject, re
 		var exists bool
 		var maxMsgSize int32
 		if s.JetStreamIsClustered() {
-			js, _ := s.getJetStreamCluster()
-			if sa := js.streamAssignment(acc.Name, cfg.Mirror.Name); sa != nil {
-				maxMsgSize = sa.Config.MaxMsgSize
-				streamSubs = sa.Config.Subjects
-				exists = true
+			if js, _ := s.getJetStreamCluster(); js != nil {
+				js.mu.RLock()
```

Comment:
This can go inside conditional below. ok to get to the pointer and check for nil without lock.

Reply:
The lock is needed around streamAssignment

Reply:
The lock is needed around streamAssignment
```
// Will lookup a stream assignment.
// Lock should be held.
func (js *jetStream) streamAssignment(account, stream string) (sa *streamAssignment) {
```